### PR TITLE
✨ Vue.js checkin

### PR DESCRIPTION
**Pull request by HerrLevin**

### CODE REVIEW

File: app/Exceptions/MissingParametersExection.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?php
```
Comment (MrKrisKrisu) [inline app/Exceptions/MissingParametersExection.php]: Is this Exception ever used? It' just includes but I don't see any usecases.

File: app/Exceptions/MissingParametersExection.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace App\Exceptions;
+
+use Exception;
+
+class MissingParametersExection extends Exception
```
Comment (MrKrisKrisu) [inline app/Exceptions/MissingParametersExection.php]:
```suggestion
class MissingParametersException extends Exception
```

File: app/Exceptions/MissingParametersExection.php [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+{
+    public function __construct(string $parameterName) {
+        $message = "Missing Parameter: " . $parameterName;
+        parent::__construct($message);
```
Comment (MrKrisKrisu) [inline app/Exceptions/MissingParametersExection.php]: Inline $message?

File: app/Http/Controllers/API/NotificationController.php [hunk 4]
```diff
@@ -11,7 +12,7 @@ class NotificationController extends ResponseController
     /**
      * Display a listing of the resource.
      *
-     * @return Response
+     * @return JsonResponse
      */
     public function index() {
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/API/NotificationController.php]: Can you change the return type for PHP it self also? :D

File: app/Http/Controllers/API/TransportController.php [hunk 5]
```diff
@@ -6,11 +6,13 @@
...
+use App\Exceptions\MissingParametersExection;
 use App\Exceptions\StationNotOnTripException;
 use App\Http\Controllers\HafasController;
 use App\Http\Controllers\TransportController as TransportBackend;
 use App\Models\HafasTrip;
 use Carbon\Carbon;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
 use Illuminate\Http\JsonResponse;
 use Illuminate\Http\Request;
 use Illuminate\Support\Facades\Auth;
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/API/TransportController.php]: Remove unused Imports

File: app/Http/Controllers/API/TransportController.php [hunk 6]
```diff
@@ -39,25 +41,20 @@ public function TrainStationboard(Request $request): JsonResponse {
         $validated = $validator->validate();
 
         try {
-            $trainStationboardResponse = TransportBackend::TrainStationboard(
+            $trainStationboardResponse = TransportBackend::getDepartures(
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/API/TransportController.php:43]: Thank you! <3

File: app/Http/Controllers/API/v1/NotificationController.php [hunk 7]
```diff
@@ -0,0 +1,33 @@
+<?php
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/API/v1/NotificationController.php:1]: Please autoformat this file.

File: app/Http/Controllers/API/v1/TransportController.php [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+        $validated = $validator->validate();
+
+        try {
+            $trainStationboardResponse = TransportBackend::getDepartures(
+                $name,
+                isset($validated['when']) ? Carbon::parse($validated['when']) : null,
+                $validated['travelType'] ?? null
+            );
+        } catch (HafasException $exception) {
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/API/v1/TransportController.php]: $exception isn't used here

File: app/Http/Controllers/API/v1/TransportController.php [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+            return $this->sendError("There has been an error with our data provider", 400);
+        } catch (ModelNotFoundException) {
+            return $this->sendError("Your query matches no station", 404);
+        }
+
+        return $this->sendv1Response(
+            data: $trainStationboardResponse['departures'],
+            additional: ["meta" => ['station' => $trainStationboardResponse['station'],
+                                    'times'   => $trainStationboardResponse['times'],
+                ]]
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/API/v1/TransportController.php:47]: Can you improve the formatting of the array?

File: app/Http/Controllers/FrontendTransportController.php [hunk 10]
```diff
@@ -6,10 +6,12 @@
...
 use App\Exceptions\HafasException;
+use App\Exceptions\MissingParametersExection;
 use App\Http\Controllers\Backend\EventController as EventBackend;
 use App\Http\Controllers\TransportController as TransportBackend;
 use Carbon\Carbon;
 use Illuminate\Contracts\Support\Renderable;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
 use Illuminate\Http\JsonResponse;
 use Illuminate\Http\RedirectResponse;
 use Illuminate\Http\Request;
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/FrontendTransportController.php]: Remove unused imports

File: app/Http/Controllers/FrontendTransportController.php [hunk 11]
```diff
@@ -40,25 +42,21 @@ public function TrainStationboard(Request $request): Renderable|RedirectResponse
...
-        if ($TrainStationboardResponse === null) {
+        } catch (ModelNotFoundException) {
             return redirect()->back()->with('error', __('controller.transport.no-station-found'));
         }
 
         return view('stationboard', [
                                       'station'    => $TrainStationboardResponse['station'],
                                       'departures' => $TrainStationboardResponse['departures'],
-                                      'when'       => $TrainStationboardResponse['when'],
+                                      'times'       => $TrainStationboardResponse['times'],
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/FrontendTransportController.php]: format pls.

File: app/Http/Controllers/TransportController.php [hunk 12]
```diff
@@ -23,6 +26,7 @@
 use GuzzleHttp\Client;
 use GuzzleHttp\Exception\GuzzleException;
 use GuzzleHttp\Exception\RequestException;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
 use Illuminate\Support\Collection;
 use Illuminate\Support\Facades\DB;
 use Illuminate\Support\Facades\Log;
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/TransportController.php:30]: Remove unused imports

File: app/Http/Controllers/TransportController.php [hunk 13]
```diff
@@ -59,48 +63,45 @@ public static function BusAutocomplete($station) {
...
+            if ($station == null) {
+                throw new ModelNotFoundException;
+            }
         }
-        if ($station == null) {
-            return null;
-        }
+
+        $when  = $when ?? Carbon::now()->subMinutes(5);
+        $times = ['now'  => $when,
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/TransportController.php]: Please add a nl between `[` and `'now'`

File: app/Http/Controllers/TransportController.php [hunk 14]
```diff
@@ -192,6 +193,7 @@ public static function sortByWhenOrScheduledWhen(array $departuresList): array {
      * @param Carbon|null $departure
      * @return array|null
      * @throws HafasException
+     * @deprecated replaced by getTrainTrip
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/TransportController.php:195]: <3

File: resources/components/NotificationsButton.vue [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+  props: {
+    notificationsCount: 0,
+    toggler: false
+  }
+};
+</script>
+
+<style scoped>
+
+</style>
```
Comment (MrKrisKrisu) [inline resources/components/NotificationsButton.vue]: nl

File: resources/components/NotificationsModal.vue [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+            console.error(error);
+          });
+    }
+  }
+};
+</script>
+
+<style scoped>
+
+</style>
```
Comment (MrKrisKrisu) [inline resources/components/NotificationsModal.vue]: nl

File: routes/api.php [hunk 17]
```diff
@@ -25,9 +26,9 @@
         Route::post('login', [v1Auth::class, 'login']);
         Route::post('signup', [v1Auth::class, 'signup']);
         Route::group(['middleware' => 'auth:api'], function() {
-            Route::post('refresh', [v1Auth::class, 'refresh']);
-            Route::post('logout', [v1Auth::class, 'logout']);
-            Route::get('user', [v1Auth::class, 'user']);
+            Route::post('refresh', [v1Auth::class, 'refresh'])->middleware('auth:api');
+            Route::post('logout', [v1Auth::class, 'logout'])->middleware('auth:api');
+            Route::get('user', [v1Auth::class, 'user'])->middleware('auth:api');
```
Comment (MrKrisKrisu) [inline routes/api.php]: middleware also specified in group?

**Review by MrKrisKrisu — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/API/v1/TransportController.php (hunk 9 above)
Comment (HerrLevin) [reply app/Http/Controllers/API/v1/TransportController.php:47]: Not yet. This still needs refactoring.

File: app/Http/Controllers/TransportController.php (hunk 12 above)
Comment (HerrLevin) [reply app/Http/Controllers/TransportController.php:30]: You know that the ModelNotFoundException is used here? :D

File: app/Http/Controllers/API/v1/TransportController.php (hunk 9 above)
Comment (MrKrisKrisu) [reply app/Http/Controllers/API/v1/TransportController.php:47]: And what's stopping you from improving the bad formatting?

File: app/Http/Controllers/TransportController.php (hunk 12 above)
Comment (MrKrisKrisu) [reply app/Http/Controllers/TransportController.php:30]: It was about MissingParametersExection which you've already removed.

**Comment by MrKrisKrisu** [comment]
Please have a look at the tests.

**Review by MrKrisKrisu — APPROVED** [review]

**Pull request merged**